kyaml/setters2/set.go

```diff
@@ -269,11 +270,40 @@ func validateAgainstSchema(ext *CliExtension, sch *spec.Schema) error {
 	}
 	err = validate.AgainstSchema(&sc, input, strfmt.Default)
 	if err != nil {
-		return errors.Errorf("The input value doesn't validate against provided OpenAPI schema: %v\n", err.Error())
+		// We know that the error returned by AgainstSchema is of type
+		// *oaerrors.CompositeError
+		compErr := err.(*oaerrors.CompositeError)
+		filteredErr := filterStringTypeErrors(compErr)
+		if filteredErr != nil {
+			return errors.Errorf("The input value doesn't validate against provided OpenAPI schema: %v\n", err.Error())
+		}
 	}
 	return nil
 }
 
+// filterStringTypeErrors filters out all errors about a value not being
+// of type string. When we turn unmarshal the values from a yaml document
+// into go types, some values will be parsed as non-string types. But they
+// are nonetheless valid values for string.
+// This will filter the errors and return a new error if there are remaining
+// validation errors after filtering, and nil if we no longer have any
+// errors.
+func filterStringTypeErrors(compErr *oaerrors.CompositeError) error {
+	var filteredErrors []error
+	for _, err := range compErr.Errors {
+		if strings.Contains(err.Error(), "in body must be of type string") {
```

Comment:
I am not sure if this should be a post step. Parsing error message for particular substring might be error-prone as if there is slight change in the library error message then this is broken. And the intent is also not very clear here. The aim here is to avoid validating string type and accept all values as valid string. This could be done as a pre-step by converting input values to string if the setter definition has string type.

Reply:
I have updated the PR to instead change the generated yaml to force the type of the value to be string whenever the type is string. I think this is a better solution than the previous one.